### PR TITLE
Move queued tracking lock implementation to core for other uses.

### PR DESCRIPTION
**Pull request by diosmosis**

### CODE REVIEW

File: core/Concurrency/LockBackend/MySqlLockBackend.php [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    public function install()
+    {
+        if (self::$tableCreated) {
+            return;
+        }
+
+        $tableName = self::getTableName();
+
+        DbHelper::createTable($tableName, "
```
Comment (tsteur) [inline core/Concurrency/LockBackend/MySqlLockBackend.php]: As the table is installed on demand... If there's a race condition, should we maybe catch a "table already exists" error and fail silently? Or we just install the table the regular way and provide an update for it as well? IT be fine if table is not needed by most users.

File: core/Concurrency/LockBackend/MySqlLockBackend.php [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+namespace Piwik\Concurrency\LockBackend;
+
+
+use Piwik\Common;
+use Piwik\Concurrency\LockBackend;
+use Piwik\Db;
+use Piwik\DbHelper;
+
+class MySqlLockBackend implements LockBackend
```
Comment (tsteur) [inline core/Concurrency/LockBackend/MySqlLockBackend.php]: Might be good to write some simple tests for this backend as well? Can be probably copied from Queued tracking

File: core/Concurrency/LockBackend/MySqlLockBackend.php (hunk 1 above)
Comment (diosmosis) [reply core/Concurrency/LockBackend/MySqlLockBackend.php]: I thought the SQL was 'CREATE IF NOT EXISTS', but I'll double check.

File: core/Concurrency/LockBackend/MySqlLockBackend.php (hunk 2 above)
Comment (diosmosis) [reply core/Concurrency/LockBackend/MySqlLockBackend.php]: Ah yes, I forgot about the backend, I'll add the tests.

File: core/Concurrency/LockBackend/MySqlLockBackend.php (hunk 1 above)
Comment (diosmosis) [reply core/Concurrency/LockBackend/MySqlLockBackend.php]: Looks like it's not `CREATE IF NOT EXISTS`, but the exception is already caught: https://github.com/matomo-org/matomo/blob/3.x-dev/core/Db/Schema/Mysql.php#L452

**Comment by diosmosis** [comment]
@tsteur Updated, though can't be merged just yet. I moved the table creation to Schema.php, since the lock could be used during tracking, and `Db::exec` tries to use the profiler.

**Pull request merged**